Intersect.Client/Entities/Player.cs

```diff
@@ -2372,6 +2376,48 @@ public void DrawOverheadInfoOnHover()
             }
         }
 
+        public override bool ShouldDrawHpBar
+        {
+            get
+            {
+                if (LatestMap == default)
+                {
+                    return false;
+                }
+
+                if (!ShouldDraw)
+                {
+                    return false;
+                }
+
+                var health = Vital[(int)Vitals.Health];
+                if (health < 1)
+                {
+                    return false;
+                }
```

Comment:
Since this is duplicated, create a `protected virtual bool ShouldNotDrawHpBar` that does these computations, and then in both `ShouldDrawHpBar` check if ShouldNotDrawHpBar return false